applications/nrf_desktop/src/modules/hid_state.c

```diff
@@ -977,6 +988,15 @@ static struct report_state *find_next_report_state(const struct report_data *rd)
 	return rs_result;
 }
 
+static void broadcast_keyboard_leds(struct subscriber *sub)
+{
+	struct keyboard_leds_event *event = new_keyboard_leds_event();
```

Comment:
I don't understand why we need this hop. The function that sets the leds would be what... 6 lines of code. It would be simpler to duplicate it if that's the concern. But I don't see what it is concern as only peripheral devices will set leds and these will include hid_state. So I don't get what we need new module and new event for that.

Reply:
We need to track subscriber for keyboard keys HID input report. We actually display keyboard LEDs related to this particular subscriber (it's why we need information from HID state to handle the output report).

Reply:
I also wanted to clearly translate the HID output report to state of the LEDs and keep functionalities separated (HID state module is already quite complex).

Reply:
>We need to track subscriber for keyboard keys HID input report.

This I understand but it is done in hid_state. After that it's just a matter of setting leds to correct state depending on a current subscriber. As mentioned I don't see any additional value in separating that into new module.